js/index.js

```diff
@@ -58,33 +74,96 @@ webix.ready(function(){
             {view:"resizer"},
             {
               view:"datatable",
+              id:"films_datatable",
               scrollY:true,
               autoConfig:true,
               minWidth:400,
               data:grid_data
             },
             {
               view:"form",
-              width:250,
+              id:"films_form",
+              width:350,
               elements:[
                 {
-                  margin:10,//margin for all elements of rows[]
+                  margin:10,
                   rows:[
                     {template:"EDIT FILMS", type:"section"},
-                    {view:"text", label:"Title"},
-                    {view:"text", label:"Year"},
-                    {view:"text", label:"Rating"},
-                    {view:"text", label:"Votes"},
+                    {view:"text", label:"Title", name:"title", invalidMessage:"Enter Title of Film"},
+                    {view:"text", label:"Year", name:"year", invalidMessage:"Enter Year between 1970 and 2021"},
+                    {view:"text", label:"Rating", name:"rating", invalidMessage:"Enter Raiting between 1 and 10"},
+                    {view:"text", label:"Votes", name:"votes", invalidMessage:"Enter Votes between 0 and 99999"},
                   ]
                 },
                 {
                   margin:20,
                   cols:[
-                    {view:"button", id:"btn_add_new", value:"Add new", css:"webix_primary"},
-                    {view:"button", id:"btn_clear", value:"Clear", css:"webix_secondary"},
+                    {view:"button", id:"btn_add_new", value:"Add new", css:"webix_primary",
+                      click:function(){
+                        if($$("films_form").validate()){
+                            var item = $$("films_form").getValues();
+
+                            item.rank = grid_data.length+1;//add correct rank to item object
+                            grid_data.push(item);//add item object to the grid_data array
+
+                            $$("films_datatable").add(item);
```

Comment:
it would be great if you can scroll to show the newly added item. if there are a lot of data or screen is small, the new item is not seen.  hint: use a special Datatable method to do it

Reply:
I have added: 
"datatable.showItem(item.id);" for scrolling to new element
"datatable.select(item.id);" for selecting of new element